src/main/telemetry/smartport.c

```diff
@@ -566,7 +355,11 @@ void handleSmartPortTelemetry(void)
         if (smartPortRxBuffer.frameId == FSSP_MSPC_FRAME) {
 
             // Pass only the payload: skip sensorId & frameId
-            handleSmartPortMspFrame(&smartPortRxBuffer);
+            if (!smartPortMspReplyPending) {
```

Comment:
This breaks the existing behaviour of SmartPort, which is, in the case of an MSP request being received, to drop the MSP request that is currently being processed, and not drop the last MSP request that has been received.

I feel it is probably better if we do not change how the protocol is implemented here.

Reply:
From what I recall, the lua scripts don't expect that their MSP requests are blocking.  The script just keeps sending multiple page requests until values are received -- only then does it create a block of sorts.  In the event that a frame is received by the TX as another outbound request is made, testing has shown that these requests often overlap causing confusion in the FC's incoming frame buffer.  The existing implementation also does not push out any sort of association of the outgoing data to the original request, so it's possible that the values could get loaded on the wrong page on the transmitter or cause unexpected local null values to crash the script.

I will try this and see what happens, but I recall changing this behavior to correct these issues.

Reply:
Looks like this could likely be handled easily by resetting mspStarted = 0 if the buffer is not empty rather than returning false.

Reply:
I tested smartport and crsf using the mspStarted = 0 fix.  Returning false will unfortunately cause the original smartport regression to happen again where it gets hung up between fast page changes.

Reply:
The fix you provided in #4120 looks good to me. My criticism wasn't so much pertaining to the known problems with the taranis telemetry script, but about changing the way the MSP over SmartPort protocol is implemented as such a late stage of the 3.2 RCs. This could break any other clients of the protocol that we don't know about, so it should be avoided.

Reply:
No problem :) I understand.  Glad we could get it working as expected.